pyiceberg/catalog/dynamodb.py

```diff
@@ -652,6 +654,19 @@ def _convert_dynamo_table_item_to_iceberg_table(self, dynamo_table_item: Dict[st
             catalog=self,
         )
 
+    def _get_default_warehouse_location(self, database_name: str, table_name: str) -> str:
```

Comment:
Looks like we're repeating this a couple of times, should we move this into `pyiceberg/catalog/__init__.py`?